types/react/test/next.tsx

```diff
@@ -36,7 +36,7 @@ function useExperimentalHooks() {
     // $ExpectType boolean
     const deferredToggle = React.useDeferredValue(toggle);
 
-    const [func] = React.useState(() => () => 0);
+    const [func] = React.useState<() => number>(() => () => 0);
```

Comment:
One last thing before we forget this: Let's add a TODO here or similar that makes it clear that we'd like to avoid the explicit type parameter here

Reply:
I was negative about adding TODO here because it's a problem with ts, not dts.
...but after I heard that, I thought it'd be better to add it! Thanks :)